astropy/nddata/nddata.py

```diff
@@ -272,6 +283,10 @@ def wcs(self):
         """
         return self._wcs
 
+    @wcs.setter
+    def wcs(self, wcs):
+        self._wcs = wcs
```

Comment:
If we have a setter we probably should just call it in the `__init__` and put the logic to validate if it's a correct instance there.

Reply:
Done